Error handling

dmahely opened the pull request:
I added a component called `<Container />` that renders `<App />` within `<ErrorBoundary />`, which in turn catches errors and renders an `<Error />` component in case any are found.

Next steps: 
1. Design nice UI for `<Error />` and implement it
2. Figure out what the 'Try again' button in it should do, I assume just refresh the page?

tryggvigy commented:
> 2. Figure out what the 'Try again' button in it should do, I assume just refresh the page?

Yeah I think that makes sense!

tryggvigy commented on src/components/Error/index.js:
```diff
@@ -0,0 +1,3 @@
+import { Error } from './Error';
+
+export { Error };
```
You should be able to do `export { Error } from './Error';` if you want

dmahely replied:
Neat! I got this trick from a post somewhere and didn't think much about improving it. I'll change the index files soon.

dmahely commented:
I spent too much time trying to draw something to put in the error page, but ended up going another way! I added the design in Figma and implemented it here. I've been using `throw new Error('whatever')` in App's useEffect method to see the error page. There's a bit of white space (24px exactly) at the top in that page only that I can't figure out where it's coming from.

tryggvigy commented on src/components/Error/Error.js:
```diff
@@ -9,7 +11,15 @@ const Error = (hasError) => {
+            <img
+                className="Error--image"
+                src={albumCover}
+                alt="Please don't stop the music song cover"
```
nice alt tag! :) those are important when it's possible to define them properly

dmahely commented:
Finally figured out where the pesky extra margin is coming from!
